service/resource/deployment/resource.go

```diff
@@ -187,7 +190,85 @@ func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desire
 }
 
 func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
-	return nil, nil, nil, nil
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	currentDeployments, err := toDeployments(currentState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	desiredDeployments, err := toDeployments(desiredState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+
+	var deploymentsToCreate interface{}
+	{
+		deploymentsToCreate, err = r.GetCreateState(ctx, obj, currentState, desiredState)
+		if err != nil {
+			return nil, nil, nil, microerror.Mask(err)
+		}
+	}
+
+	var deploymentsToDelete []*v1beta1.Deployment
+	{
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out which deployments have to be deleted")
+
+		for _, currentDeployment := range currentDeployments {
+			if !containsDeployment(desiredDeployments, currentDeployment) {
+				deploymentsToDelete = append(deploymentsToDelete, currentDeployment)
+			}
+		}
+
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", fmt.Sprintf("found %d deployments that have to be deleted", len(deploymentsToDelete)))
+	}
+
+	var deploymentsToUpdate []*v1beta1.Deployment
+	if updateallowedcontext.IsUpdateAllowed(ctx) {
```

Comment:
With this PR we are implementing deployment updates, but we are not allowing them yet. This is a safety net.  The blocking mechanism here is the same as we had before for the update state of the config map resource. We will remove this blocking condition later as soon as we are able to automatically update. For now the guest cluster admin has to drain guest cluster nodes and delete pods on his own. This is a first step towards updating KVM guest clusters. Using this process we enable the customer and ourselves to update guest clusters on KVM while maintaining enough operational control to not fuck everything at the same time.